go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -1028,8 +1028,14 @@ func (nc *DefaultNodeNetworkController) reconcileConntrackUponEndpointSliceEvent
 					continue
 				}
 				// upon update and delete events, flush conntrack only for UDP
-				err := util.DeleteConntrack(oldIPStr, *oldPort.Port, *oldPort.Protocol, netlink.ConntrackReplyAnyIP, nil)
-				if err != nil {
+				if err := util.ValidatePort(*oldPort.Protocol, *oldPort.Port); err != nil {
+					klog.V(5).Infof("Skipping conntrack deletion for endpoint, invalid port. "+
+						"IP: %s, port: %s, err: %q",
+						oldIPStr, *oldPort.Protocol, *oldPort.Port, err)
+					continue
+				}
+				if err := util.DeleteConntrack(oldIPStr, *oldPort.Port, *oldPort.Protocol, netlink.ConntrackReplyAnyIP,
```

Comment:
@andreaskaris could we instead create a new variant in pkg/util/net_linux.go called:

```
func DeleteConntrackServicePort(ip string, port int32, protocol kapi.Protocol, ipFilterType netlink.ConntrackFilterType, labels [][]byte) error {
	if err := ValidatePort(protocol, port); err != nil {
		klog.V(5).Infof("Skipping conntrack deletion for VIP %q, invalid port %d: %v", ip, port, err)
		return nil
	}
	return DeleteConntrack(ip, port, protocol, ipFilterType, labels)
}
```

and then just call it from all 3 places?

Reply:
Changed:
~~~
[akaris@workstation repro]$ oc logs -n openshift-ovn-kubernetes ovnkube-node-klpfv -c ovnkube-node | grep Skipping | tail
E0609 19:19:54.958322 1212610 gateway_iptables.go:457] Skipping service: test1, invalid service NodePort: invalid port number: 0
E0609 19:19:54.958327 1212610 gateway_iptables.go:457] Skipping service: test1, invalid service NodePort: invalid port number: 0
I0609 19:19:54.990804 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "172.19.55.99", protocol "TCP", port "0", err: "invalid port number: 0"
I0609 19:19:54.990819 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "192.168.123.10", protocol "TCP", port "0", err: "invalid port number: 0"
I0609 19:19:54.990823 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "192.168.124.10", protocol "TCP", port "0", err: "invalid port number: 0"
I0609 19:19:54.990827 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "192.168.125.10", protocol "TCP", port "0", err: "invalid port number: 0"
I0609 19:19:54.990830 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "192.168.18.22", protocol "TCP", port "0", err: "invalid port number: 0"
I0609 19:19:54.990834 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "2600:52:7:18:32d0:42ff:fed9:b87a", protocol "TCP", port "0", err: "invalid port number: 0"
I0609 19:19:54.990838 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "2600:52:7:18::22", protocol "TCP", port "0", err: "invalid port number: 0"
I0609 19:19:54.990841 1212610 net_linux.go:480] Skipping conntrack deletion for VIP "fc00:123::10", protocol "TCP", port "0", err: "invalid port number: 0"
~~~